cli/lib/archive.ts

```diff
@@ -55,3 +78,33 @@ export async function cleanup( archivePath: string ): Promise< void > {
 		}, 0 );
 	} );
 }
+
+async function getSymlinks(
+	dir: string
+): Promise< { isDirectory: boolean; symbolicPath: string; realPath: string }[] > {
+	const files = await fs.promises.readdir( dir );
+	const results = await Promise.all(
+		files.map( async ( file ) => {
+			const filePath = path.join( dir, file );
+			// Using lstat to use isSymbolicLink method, see https://nodejs.org/api/fs.html#statsissymboliclink
+			const stats = await fs.promises.lstat( filePath );
+
+			if ( stats.isSymbolicLink() ) {
+				const realPath = await fsPromises.realpath( filePath );
+				const realPathStats = await fsPromises.stat( realPath );
+				return [ { isDirectory: realPathStats.isDirectory(), symbolicPath: filePath, realPath } ];
+			}
+
+			if ( stats.isDirectory() ) {
+				return await getSymlinks( filePath );
+			}
+			// Regular file
+			return [];
+		} )
+	);
+	return results.flat();
+}
+
+function shouldExcludeEntry( entryName: string ): boolean {
```

Comment:
Have you considered calling `shouldExcludeEntry` in the `getSymlinks` function?
I think it could save us some execution time.

Reply:
Good point, I didn't think about that in the symlinks, but it's worth filtering out those, let me add it.

Reply:
Suggestion added here [5ac3d81](https://github.com/Automattic/studio/pull/1427/commits/5ac3d81e9f9fdfd363f8b64e4a476a43ac83f610)